docs/io/ascii/read.rst

```diff
@@ -793,9 +793,11 @@ the :ref:`io_registry` (note that our format has no mechanism to write out the u
 .. testcleanup::
 
     >>> from astropy.io import registry
+    >>> from astropy.io.ascii.core import FORMAT_CLASSES
     >>> for format_name in ['custom_no_header', 'custom_commented_header', 'fixed_width_commented_header']:
     ...     registry.unregister_reader(f"ascii.{format_name}", Table)
     ...     registry.unregister_writer(f"ascii.{format_name}", Table)
+    ...     del FORMAT_CLASSES[format_name]
```

Comment:
@taldcroft - it is probably not intended that format classes stay around even if you unregister the reader and the writer. It may not matter too much in practice, but without removing it, I get a failure in https://github.com/astropy/astropy/blob/11aa662ac25f6ecfc2b5f2d7efc98b197e8801f0/astropy/table/tests/test_mixin.py#L111-L123

Note that this uses a private method, so a user might never hit this problem.

Reply:
The `io.ascii` `FORMAT_CLASSES` is formally independent of registration as a `Table` reader/writer since users may be calling `io.ascii.read(...)`. In the current paradigm, unregistering should not imply removal from `FORMAT_CLASSES`.

In theory we could decide to make `io.ascii` only be providing connect functions for unified I/O, and not provide any public API. But that would be a lot of churn without really helping all that much.

Reply:
OK, makes sense. I wondered about exposing some of this (also the units registry list), but think in the end it is not super useful and certainly not urgent: in general, I think people are pretty used to restarting python if something really gets messy with registered classes, etc.